java/core/src/java/org/apache/orc/impl/IOUtils.java

```diff
@@ -30,6 +30,11 @@ public final class IOUtils {
 
   public static final int DEFAULT_BUFFER_SIZE = 8192;
 
+  /**
+   * The maximum size of array to allocate, value being the same as {@link java.util.Hashtable}
```

Comment:
Can we add info like max array to allocate for what purpose? Please only add if the specific context is known, else we leave it as is.

Reply:
https://github.com/openjdk/jdk/blob/jdk-17%2B35/src/java.base/share/classes/java/util/Hashtable.java#L391-L397

Because the comment for `java.util.Hashtable#MAX_ARRAY_SIZE` is detailed, I did not add excessive description.